xmppserver/src/main/java/org/jivesoftware/openfire/muc/MUCRole.java

```diff
@@ -444,4 +674,137 @@ public static Affiliation valueOf(int value) {
             }
         }
     }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
```

Comment:
Possibly more easily written as `return Objects.hashCode(nick, rJid, roomJid, userJid);`

Reply:
The code as-is is generated by the IDE. I trust it implicitly more than anything we've reasoned over ourselves. :)